Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -3069,6 +3069,8 @@ static WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     // Grant the tab a temporary permission to access to a pattern matching the current URL.
     RefPtr pattern = WebExtensionMatchPattern::getOrCreate(currentURL);
     tab.setTemporaryPermissionMatchPattern(pattern.copyRef());
+    if (pattern)
+        permissionsDidChange(WKWebExtensionContextPermissionMatchPatternsWereGrantedNotification, MatchPatternSet { *pattern });
```

Comment:
Can you file a bug about passing `tab` to `permissionsDidChange()` so it can be included in the notification and add as a FIXME comment here? I do worry this could be confusing without including it, making it seem like broader permissions were granted.